src/app/features/nonce/use-persisted-nonce.tsx

```diff
@@ -0,0 +1,41 @@
+import { useState } from 'react';
+
+import { isDefined, isUndefined } from '@shared/utils';
+
+import { useOnMount } from '@app/common/hooks/use-on-mount';
+import { useNextNonce } from '@app/query/stacks/nonce/account-nonces.hooks';
+
+export function usePersistedNonce() {
```

Comment:
Per state recommendation

```suggestion
export function usePersistedNonce(stacksAddress: string) {
```